modules/imgproc/src/shapedescr.cpp

```diff
@@ -454,6 +455,362 @@ cv::RotatedRect cv::fitEllipse( InputArray _points )
     return box;
 }
 
+cv::RotatedRect cv::fitEllipseAMS( InputArray _points )
+{
+    Mat points = _points.getMat();
+    int i, n = points.checkVector(2);
+    int depth = points.depth();
+    CV_Assert( n >= 0 && (depth == CV_32F || depth == CV_32S));
+
+    RotatedRect box;
+
+    if( n < 5 )
+        CV_Error( CV_StsBadSize, "There should be at least 5 points to fit the ellipse" );
+
+    Point2f c(0,0);
+
+    bool is_float = depth == CV_32F;
+    const Point* ptsi = points.ptr<Point>();
+    const Point2f* ptsf = points.ptr<Point2f>();
+
+    AutoBuffer<double> _Ad(n*6);
+    double *Ad = _Ad;
+    Mat A( n, 6, CV_64F, Ad );
+
+    AutoBuffer<double> _DMd(6*6);
+    double *DMd = _DMd;
+    Matx<double, 6, 6> DM = Matx<double, 6, 6>(DMd);
```

Comment:
Replace these 3 lines to this one:
```
Matx<double, 6, 6> DM;
```

Apply similar changes to other "fixed size" cases.